### ISSUE TITLE
Reflect cannot serialize `bevy_a11y::AccessibilityRequested`

### ISSUE THREAD

**Issue by codewing** [bevyengine/bevy#22816]
## Bevy version and features

- `0.18.0`
- `2d`, `dev`

## What you did

I'm following the scene serialization tutorial here: https://taintedcoders.com/bevy/scenes
(currently just the saving of a scene)

## What went wrong

I'm trying to save a snapshot of the world like this

```
fn save_snapshot(world: &mut World, elapsed_time: u64, run_name: String) {
    let scene = DynamicScene::from_world(world);

    let type_registry = world.resource::<AppTypeRegistry>();
    let type_registry = type_registry.read();
    let serialized_scene = scene.serialize(&type_registry).unwrap();

    // Showing the scene in the console
    info!("{}", serialized_scene);

    #[cfg(not(target_arch = "wasm32"))]
    IoTaskPool::get()
        .spawn(async move {
            use std::{fs::File, io::Write};

            File::create(format!("snapshots/{run_name}-{elapsed_time}.ron"))
                .and_then(|mut file| file.write(serialized_scene.as_bytes()))
                .expect("Error while writing scene to file");
        })
        .detach();
}
```

But this leads to a panic

```
thread 'main' (83718) panicked at testbench/src/snapshots/plugin.rs:32:60:
called `Result::unwrap()` on an `Err` value: Message("type `bevy_platform::sync::Arc<core::sync::atomic::AtomicBool>` did not register the `ReflectSerialize` or `ReflectSerializeWithRegistry` type data. For certain types, this may need to be registered manually using `register_type_data` (stack: `bevy_a11y::AccessibilityRequested` -> `bevy_platform::sync::Arc<core::sync::atomic::AtomicBool>`)")
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Encountered a panic when applying buffers for system `testbench::order::order_manager::complete_orders`!
Encountered a panic in system `bevy_app::main_schedule::Main::run_main`!
```

## Additional information

To me it looks like the `bevy_a11y::AccessibilityRequested` doesn't support reflection and would need a reflect ignore or something appropriate

**Comment by codewing** [comment]
Here is a minimal example if you want to test it out [minimal_repro.zip](https://github.com/user-attachments/files/25177972/minimal_repro.zip)

~~I assume the solution would be to correctly serialize the `bevy_platform::sync::Arc<core::sync::atomic::AtomicBool>` type or while we are at it the whole bevy_platform::sync::Arc?~~

I found that this seems to done here: `crates/bevy_reflect/src/impls/bevy_platform/sync.rs`

Not sure what's missing - if someone can give me a hint where to start I'm happy to contribute

**Comment by codewing** [comment]
After coming back to this over and over again I gave it another look

---

I did ignore AccessibilityRequested and some struct in Time (i think it was Real) and then it was able to serialize the whole world.
Looking at this it's definitely not what I want. It stores my current display configuration as well which I'm not really interested in.

---

Beyond the default from the tutorial there is another way of serializing the scene with more control that I didn't spot before: `DynamicSceneBuilder` (i think in 0.19+ it will be `DynamicWorldBuilder` in case you're reading this in the future)

The idea is to explicitly allow and deny "stuff" so I just added the resources and components that I want to have in my save files.

```rust
use bevy::prelude::*;

fn main() -> AppExit {
    App::new()
        .add_plugins(DefaultPlugins)
        .register_type::<MySaveData>()
        .register_type::<WorldInfo>()
        .insert_resource(WorldInfo {
            level: "test".to_string(),
        })
        .add_systems(Startup, init)
        .add_systems(Update, space_to_save)
        .run()
}

fn init(mut commands: Commands) {
    commands.spawn(MySaveData {
        test: "Hej!".to_string(),
    });
}

fn space_to_save(mut commands: Commands, keys: Res<ButtonInput<KeyCode>>) {
    if keys.just_pressed(KeyCode::Space) {
        commands.queue(save_scene_system);
    }
}

#[derive(Component, Reflect)]
#[reflect(Component)]
pub struct MySaveData {
    test: String,
}

#[derive(Resource, Reflect)]
#[reflect(Resource)]
pub struct WorldInfo {
    level: String,
}

fn save_scene_system(world: &mut World) {
    let mut query = world.query::<Entity>();
    let entities: Vec<Entity> = query.iter(world).collect();

    let type_registry = world.resource::<AppTypeRegistry>().clone();
    let registry = type_registry.read();

    let scene = DynamicSceneBuilder::from_world(world)
        .deny_all()
        .allow_component::<MySaveData>()
        .allow_component::<Transform>()
        .allow_component::<Visibility>()
        .allow_resource::<WorldInfo>()
        .extract_entities(entities.into_iter())
        .extract_resources()
        .remove_empty_entities()
        .build();
    let serialized_scene = scene.serialize(&registry).unwrap();

    info!("{}", serialized_scene);
}
```

I would be fine with closing this but let the team/community decide
